src/bin/ion/commands/beta/inspect.rs

```diff
@@ -184,106 +191,32 @@ fn inspect_file(input_file_name: &str,
-    output: OutputRef,
-    text_buffer: String,
-}
-
-impl SystemLevelEventSummarizer {
-    pub fn new(output: OutputRef) -> SystemLevelEventSummarizer {
-        SystemLevelEventSummarizer {
-            output,
-            text_buffer: String::with_capacity(512),
-        }
-    }
-}
-
 const IVM_HEX: &str = "e0 01 00 ea";
 const IVM_TEXT: &str = "// Ion 1.0 Version Marker";
 // System events (IVM, symtabs) are always at the top level.
 const SYSTEM_EVENT_INDENTATION: &str = "";
-
-impl SystemEventHandler for SystemLevelEventSummarizer {
-    // TODO: At the moment, the SystemEventHandler trait's functions do not have a return type that
-    //       would allow errors to bubble up. If writing to the output stream fails for some
-    //       reason, the program will end and a more terse error message will be displayed.
-    //       See: https://github.com/amzn/ion-rust/issues/118
-    fn on_ivm(&mut self, _ion_version: (u8, u8)) {
-        output(
-            &self.output,
-            None,
-            None,
-            SYSTEM_EVENT_INDENTATION,
-            IVM_HEX,
-            IVM_TEXT.dimmed(),
-        ).expect("output() failure from on_ivm()");
-    }
-
-    fn on_symbol_table_append(&mut self, symbol_table: &SymbolTable, starting_id: usize) {
-        self.text_buffer.clear();
-        self.text_buffer.push_str("// Local symbol table append: [\"");
-        join_into(&mut self.text_buffer, "\", \"", symbol_table.symbols_tail(starting_id).iter());
-        self.text_buffer.push_str("\"]");
-        output(
-            &self.output,
-            None,
-            None,
-            SYSTEM_EVENT_INDENTATION,
-            "...",
-            &self.text_buffer.dimmed(),
-        ).expect("output() failure from on_symbol_table_append()");
-    }
-
-    fn on_symbol_table_reset(&mut self, symbol_table: &SymbolTable) {
-        const ION_1_0_SYSTEM_TABLE_LENGTH: usize = 10;
-        self.text_buffer.clear();
-        if symbol_table.len() > ION_1_0_SYSTEM_TABLE_LENGTH {
-            self.text_buffer.push_str("// New local symbol table: [\"");
-            join_into(&mut self.text_buffer, "\", \"", symbol_table.symbols_tail(ION_1_0_SYSTEM_TABLE_LENGTH).iter());
-            self.text_buffer.push_str("\"]");
-        } else {
-            self.text_buffer.push_str("// Using system symbol table");
-        }
-
-        output(
-            &self.output,
-            None,
-            None,
-            SYSTEM_EVENT_INDENTATION,
-            "...",
-            &self.text_buffer.dimmed(),
-        ).expect("output() failure from on_symbol_table_reset()");
-    }
-}
-
 const LEVEL_INDENTATION: &str = "  "; // 2 spaces per level
 const TEXT_WRITER_INITIAL_BUFFER_SIZE: usize = 128;
 
-struct IonInspector<'input> {
-    output: OutputRef,
-    reader: Reader<BinaryIonCursor<io::Cursor<&'input [u8]>>>,
+struct IonInspector<'a> {
+    output: &'a mut OutputRef,
+    reader: SystemReader<RawBinaryReader<io::Cursor<&'a [u8]>>>,
```

Comment:
The `IonInspector` now has a `SystemReader` instead of a user-level `Reader` + event handler for system data.